### ISSUE TITLE
Cannot install CLI due to dependency conflict

### ISSUE THREAD

**Issue by Chriscbr** [cdk8s-team/cdk8s-cli#71]
Error:

```
<git:(master*)> npm install -g cdk8s-cli
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: cdk8s-plus-22@1.0.0-beta.4
npm WARN Found: cdk8s@1.0.0-beta.46
npm WARN node_modules/cdk8s-cli/node_modules/cdk8s
npm WARN   cdk8s@"1.0.0-beta.46" from cdk8s-cli@1.0.0-beta.53
npm WARN   node_modules/cdk8s-cli
npm WARN     cdk8s-cli@"*" from the root project
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer cdk8s@"1.0.0-beta.50" from cdk8s-plus-22@1.0.0-beta.4
npm WARN node_modules/cdk8s-cli/node_modules/cdk8s-plus-22
npm WARN   cdk8s-plus-22@"^1.0.0-beta.0" from cdk8s-cli@1.0.0-beta.53
npm WARN   node_modules/cdk8s-cli
npm WARN 
npm WARN Conflicting peer dependency: cdk8s@1.0.0-beta.50
npm WARN node_modules/cdk8s
npm WARN   peer cdk8s@"1.0.0-beta.50" from cdk8s-plus-22@1.0.0-beta.4
npm WARN   node_modules/cdk8s-cli/node_modules/cdk8s-plus-22
npm WARN     cdk8s-plus-22@"^1.0.0-beta.0" from cdk8s-cli@1.0.0-beta.53
npm WARN     node_modules/cdk8s-cli

changed 180 packages, and audited 189 packages in 2s

47 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

**Comment by Chriscbr** [comment]
A temporary workaround is to install a different version. This currently is working for me:

`npm install -g cdk8s-cli@1.0.0-beta.52`

**Comment by Chriscbr** [comment]
Two proposed changes:

1. Change `cdk8s-plus-XX` to model `cdk8s` and `constructs` as caret peer dependencies, with pinned dev dependencies. This does not fix the bug, but I think this more correctly models our dependency relationships (i.e. `cdk8s-plus-XX` should be compatible with newer minor/patch versions of `cdk8s`).

2. Change `cdk8s-cli` to require `cdk8s` as a caret peer dependency instead of a pinned peer dependency. This should resolve the dependency conflict since both `cdk8s-cli` and `cdk8s-plus-22` will both allow the latest version of `cdk8s` as their dependency.

**Comment by Chriscbr** [comment]
Fixed in cdk8s-cli v1.0.0-beta.54.
